x-pack/solutions/security/packages/kbn-scout-security/src/playwright/fixtures/worker/apis/entity_analytics.ts

```diff
@@ -0,0 +1,291 @@
+        log,
+        'security.entityAnalytics.deleteEntityStoreEngines',
+        async () => {
+          await kbnClient.request({
+            method: 'DELETE',
+            path: `${basePath}${ENTITY_STORE_ENGINES_URL}`,
+            query: {
+              delete_data: 'true',
+            },
+            ignoreErrors: [404, 500],
+          });
+          // Wait for cleanup to complete - ensure server state is fully cleaned up
+          await service.waitForEntityStoreCleanup();
+        }
+      );
+    },
+
+    deleteRiskEngineConfiguration: async () => {
+      await measurePerformanceAsync(
+        log,
+        'security.entityAnalytics.deleteRiskEngineConfiguration',
+        async () => {
+          const findResponse = await kbnClient.request<{
+            saved_objects?: Array<{ id: string }>;
+          }>({
+            method: 'GET',
+            path: `${basePath}${SAVED_OBJECTS_FIND_URL}`,
+            query: {
+              type: RISK_ENGINE_CONFIGURATION_TYPE,
+            },
+            ignoreErrors: [404],
+          });
+
+          const savedObjectId = findResponse?.data?.saved_objects?.[0]?.id;
+          if (savedObjectId) {
+            await kbnClient.request({
+              method: 'DELETE',
+              path: `${basePath}/api/saved_objects/${RISK_ENGINE_CONFIGURATION_TYPE}/${savedObjectId}`,
+              ignoreErrors: [404],
+            });
+            // Wait for cleanup to complete - ensure server state is fully cleaned up
+            await service.waitForRiskEngineCleanup();
+          } else {
+            // No saved object to delete, check if already cleaned up
+            const status = await service.getRiskEngineStatus();
+            if (status.risk_engine_status !== 'NOT_INSTALLED') {
+              // Status is not clean, wait for it to become clean (might be in transition)
+              await service.waitForRiskEngineCleanup();
+            }
+          }
+        }
+      );
+    },
+
+    getRiskEngineStatus: async () => {
+      return measurePerformanceAsync(
+        log,
+        'security.entityAnalytics.getRiskEngineStatus',
+        async () => {
+          const response = await kbnClient.request<RiskEngineStatusResponse>({
+            method: 'GET',
+            path: `${basePath}${RISK_ENGINE_STATUS_URL}`,
+            headers: {
+              'elastic-api-version': API_VERSIONS.internal.v1,
+            },
+          });
+          return response.data;
+        }
+      );
+    },
+
+    getEntityStoreStatus: async () => {
+      return measurePerformanceAsync(
+        log,
+        'security.entityAnalytics.getEntityStoreStatus',
+        async () => {
+          const response = await kbnClient.request<GetEntityStoreStatusResponse>({
+            method: 'GET',
+            path: `${basePath}${ENTITY_STORE_STATUS_URL}`,
+            headers: {
+              'elastic-api-version': API_VERSIONS.public.v1,
+            },
+          });
```

Comment:
nit (just as experiment): maybe it will be easier to use `retries` prop instead of while loop in 2 places? 

This is now request retry is implemented in KbnClient:

https://github.com/elastic/kibana/blob/f1284467fcb6ffcbb8699ecd118918ecf9c2a4b4/src/platform/packages/shared/kbn-test/src/kbn_client/kbn_client_requester.ts#L133-L157